tests/integration/tier0/latest-kernel-check/test_latest_kernel_check.py

```diff
@@ -46,3 +48,47 @@ def test_verify_latest_kernel_check_passes_with_failed_repoquery(shell, convert2
     if "centos-8.5" in get_system_release:
         assert shell(f"rm -f {centos_custom_reposdir}/centos-8.5/{repofile}.repo").returncode == 0
     assert shell(f"rm -f /etc/yum.repos.d/{repofile}.repo").returncode == 0
+
+
+@pytest.mark.yum_exclude_kernel
+def test_latest_kernel_check_with_exclude_kernel_option(shell, convert2rhel):
+    """
+    Define `exclude=kernel` in /etc/yum.conf and verify, the conversion is not inhibited with:
+    'CRITICAL - Could not find any kernel from repositories to compare against the loaded kernel.'
+    """
+
+    yum_config = "/etc/yum.conf"
+    backup_dir = "/tmp/config-backup"
+    config = configparser.ConfigParser()
+    config.read(yum_config)
+    exclude_option = r"exclude=.*kernel kernel-core\n"
+
+    assert shell(f"mkdir {backup_dir}").returncode == 0
+
+    assert shell(f"cp {yum_config} {backup_dir}").returncode == 0
+    # If there is already an `exclude` section, append to the existing value
+    if config.has_option("main", "exclude"):
+        pre_existing_value = config.get("main", "exclude")
+        config.set("main", "exclude", f"{pre_existing_value} kernel kernel-core")
+    else:
+        config.set("main", "exclude", "kernel kernel-core")
+
+    with open(yum_config, "w") as configfile:
+        config.write(configfile, space_around_delimiters=False)
+
+    assert re.search(exclude_option, shell(f"cat {yum_config}").output)
```

Comment:
Can we use the config.read() here instead of the shell call? Maybe file.read() will look better here

Reply:
+1 to that

Reply:
I'm a bit struggling with matching the regex with file.read(), any suggestions on how to approach that?
@r0x0d @kokesak